### PR TITLE
Add release details and bump the jenkins lib version

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Adds release details to releasing.md.
Also bumps the jenkins lib version to 1.1.1 that includes a fix to publish the release on Github.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nhtruong** [comment]
We're waiting for the [version bump PR](https://github.com/opensearch-project/opensearch-js/pull/318) to come in first before we're merging anything else.

File: RELEASING.md [hunk 1]
```diff
@@ -31,4 +31,5 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
 
 ## Releasing
 
-The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
+The release process includes a [maintainer](MAINTAINERS.md) pushing a tag to this repository. This creates a draft release which then triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-js-release/) and releases the client on [npmjs](https://www.npmjs.com/package/@opensearch-project/opensearch).
```
Comment (VachaShah) [inline RELEASING.md:34]: @gaiksaya Should we add steps like [this](https://github.com/opensearch-project/opensearch-java/blob/main/RELEASING.md#releasing)

Comment (gaiksaya) [reply RELEASING.md:34]: Sounds good. Let me change that.

Comment (gaiksaya) [reply RELEASING.md]: Done!

File: RELEASING.md [hunk 2]
```diff
@@ -31,4 +31,10 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
 
 ## Releasing
 
-The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+1. Create a tag, e.g. v2.1.0, and push it to the GitHub repo.
```
Comment (VachaShah) [inline RELEASING.md:36]: *Nit*: Fix the numberings?

Comment (gaiksaya) [reply RELEASING.md:36]: Its in markdown. If you click on view file you will see the numbering auto corrected. https://www.markdownguide.org/basic-syntax/#ordered-lists

Comment (VachaShah) [reply RELEASING.md:36]: My bad 🤦🏼‍♀️

**Review by VachaShah — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**